core/src/main/java/org/eclipse/hono/util/BasicExpiringValue.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ * A base class for implementing an expiring value.
+ *
+ * @param <T> The value type.
+ */
+public class BasicExpiringValue<T> implements ExpiringValue<T> {
+
+    private T value;
+    private Instant expirationTime;
+
+    /**
+     * Creates a new instance for a value and an expiration time.
+     * 
+     * @param value The value.
+     * @param expirationTime The instant after which the value will be considered expired.
+     */
+    public BasicExpiringValue(final T value, final Instant expirationTime) {
+        this.value = Objects.requireNonNull(value);
+        this.expirationTime = Objects.requireNonNull(expirationTime);
+    }
+
+    @Override
+    public final T getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean isExpired() {
+        return isExpired(Instant.now());
+    }
+
+    @Override
+    public boolean isExpired(final Instant now) {
```

Comment:
`now` seems to be not the best name for the parameter FMPOV.
How about renaming it to `instant`? Or to `referencePointInTime`?

And: do we need to have this method in the interface? Or can it be made purely internal (to implement `isExpired()`)?

Reply:
Yes, we should rename the parameter and I would like to keep the method in the interface even if just for supporting easier testing ...